nuvlaedge/agent/infrastructure.py

```diff
@@ -236,7 +238,7 @@ def commission_vpn(self):
             'vpn_shared_key': vpn_conf_fields['vpn-shared-key'],
             'vpn_common_name_prefix': vpn_conf_fields['vpn-common-name-prefix'],
             'vpn_endpoints_mapped': vpn_conf_fields['vpn-endpoints-mapped'],
-            'vpn_interface_name': self.vpn_interface_name,
+            'vpn_interface_name': derived_vpn_int_name,
```

Comment:
This change is not needed.

The default value for the VPN interface name is `tun` (at the level of NuvlaEdge agent):
https://github.com/nuvlaedge/nuvlaedge/blob/c666307f9e702d58f9234f61eac7094c5d2c81a0/nuvlaedge/agent/common/nuvlaedge_common.py#L31

When set to `tun`, OpenVPN will automatically name the interface so that it doesn't conflict with an existing one (`tun0`, `tun1`, `tun2,` ...). 
From [OpenVPN doc](https://openvpn.net/community-resources/reference-manual-for-openvpn-2-4/):
```
--dev tunX | tapX | null
    TUN/TAP virtual network device ( X can be omitted for a dynamic device.)See examples section below for an example on setting up a TUN device.
```

In practice, the interface name is `vpn` because that's the default value in the [docker-compose.yml](https://github.com/nuvlaedge/deployment/blob/main/docker-compose.yml#L64) and in the [helm chart](https://github.com/nuvlaedge/deployment/blob/main/helm/templates/agent-deployment.yaml#L56-L57).

So if you want the interface name to be dynamically defined, you can simply set the env var `VPN_INTERFACE_NAME` to `tun` when deploying/updating NuvlaEdge.

⚠️ be careful because having multiples NE on the same device with a VPN each, will not work properly if they are all connected the the NuvlaEdge VPN. 
It will work only on one vpn interface because of routing issues. Reaching the device will work but on the way back, the packets will be sent on the wrong interface.

Reply:
OK. Yup. See the logic now... will revert this change and test the behaviour.